lib/dox/printers/document_printer.rb

```diff
@@ -1,26 +1,68 @@
 module Dox
   module Printers
     class DocumentPrinter < BasePrinter
+      def initialize(output)
+        super(acquire_body(api_info_path))
+        @output = output
+      end
+
       def print(passed_examples)
-        print_meta_info
+        spec['paths'] = {}
+        spec['tags'] = []
+        spec['x-tagGroups'] = []
 
         passed_examples.sort.each do |_, resource_group|
           group_printer.print(resource_group)
         end
+
+        @output.puts(JSON.pretty_generate(spec))
       end
 
       private
 
-      def group_printer
-        @group_printer ||= ResourceGroupPrinter.new(@output)
+      def acquire_body(body, fullpath = false)
+        return if body.blank?
+
+        if body.to_s =~ /.*\.json$/
+          body = content(acquire_path(body, fullpath))
+          adjust_description(body['info'])
+
+          return body
+        end
+
+        raise Dox::Errors::FileNotFoundError, 'No such .json file was found.'
+      end
+
+      def acquire_path(desc, fullpath = false)
+        if fullpath
+          desc
+        else
+          descriptions_folder_path.join(desc).to_s
+        end
+      end
+
+      def adjust_description(info)
+        info['description'] = acquire_desc(info['description']) if info['description'].end_with?('.md')
+      end
+
+      def acquire_desc(path)
+        File.read(File.join(Dox.config.desc_folder_path, path))
```

Comment:
this might be a separate, global function as you are using `File.read(File.join` in multiple places